analysis/filter/StructureModifier/src/main/kotlin/de/maibornwolff/codecharta/filter/structuremodifier/NodeRemover.kt

```diff
@@ -29,18 +29,23 @@ class NodeRemover(private val project: Project) {
         ).build(cleanAttributeDescriptors = true)
     }
 
-    private fun filterNodes(path: List<String>, node: MutableNode): MutableNode {
-        val filteredChildren = node.children.filter { it.name != path.firstOrNull() || path.size > 1 }
-        node.children = filteredChildren.map {
-            filterNodes(path.drop(1), it)
-        }.toMutableSet()
-        return node
-    }
+    private fun removeNodes(paths: List<List<String>>): List<MutableNode> {
+        val rootNode = project.rootNode.toMutableNode()
+
+        for (path in paths) {
+            var currentNode = rootNode
+            val pathWithoutRoot = path.takeLastWhile { it != "root" }
```

Comment:
What happens, when have a project like `/root/src/project1/root/src` and `/root/src/project1/root` or  `/root/src/project1/root/src` should be removed?

Reply:
Thanks for the catch! 
This was meant to solve the problem of gitbash converting paths starting with forward slash to absolute paths.
While other projects ignore this edge case (https://github.com/moby/moby/issues/24029), I'd suggest throwing a warning if the given path doesn't start with "root".
Other than that I don't see a quick solution to handling this correctly.